pkg/store/cache.go

```diff
@@ -1,47 +1,159 @@
 package store
 
 import (
-	lru "github.com/hashicorp/golang-lru"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/simplelru"
 	"github.com/oklog/ulid"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/tsdb/labels"
 )
 
+const (
+	cacheTypePostings = "postings"
+	cacheTypeSeries   = "series"
+)
+
 type cacheItem struct {
 	block ulid.ULID
 	key   interface{}
 }
+
+func (c cacheItem) keyType() string {
+	switch c.key.(type) {
```

Comment:
How does it work?

https://play.golang.org/p/tHmYrn-SsR

Reply:
Your playground code checks the `cacheType*` string. Here we check `cacheKey*`.

Reply:
ah did see the `Key` word there, ok